ComplementaryScripts/modelCuration/AddCofactorIon.m

```diff
@@ -0,0 +1,216 @@
+function model = AddCofactorIon(model,data)
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% addBiomassUpdate
+% Adds rxns and metabolites for BiomassUpdate.
+% Input: model, Biomass_newRxnMatrix.tsv, Biomass_newRxnProp.tsv,
+%        Biomass_newRxnMet.tsv.
+%
+% NOTE: changeGeneAssociation.m is a function from COBRA
+%
+% Feiran Li     2018-10-23
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+
+% Load stoichiometry data:
+cd ..
+fid = fopen('../ComplementaryData/modelCuration/Biomass_newRxnMatrix.tsv');
+newreaction = textscan(fid,'%s %s %s %s %s','Delimiter','\t','HeaderLines',1);
+matrix.rxnIDs  = newreaction{1};
+matrix.metcoef = cellfun(@str2num, newreaction{2});
+matrix.metIDs  = newreaction{3};
+matrix.mettype = newreaction{4};
+matrix.metcompartments = newreaction{5};
+fclose(fid);
+
+% Load rxn properties data:
+fid  = fopen('../ComplementaryData/modelCuration/Biomass_newRxnProp.tsv','r');
+rev = textscan(fid,'%s %s %s %s %s %s %s','Delimiter','\t','HeaderLines',1);
+newrxn.ID  = rev{1};
+newrxn.Rev = cellfun(@str2num, rev{2});
+newrxn.GPR = rev{3};
+newrxn.rxnNames      = rev{4};
+newrxn.rxnECNumbers  = rev{5};
+newrxn.rxnKEGGID     = rev{6};
+newrxn.rxnNotes      = rev{7};
+newrxn.rxnMetaNetXID = newrxn.ID;
+for i = 1:length(newrxn.rxnMetaNetXID)
+    if ~startsWith(newrxn.rxnMetaNetXID{i},'MNXR')
+        newrxn.rxnMetaNetXID{i} = '';
+    end
+end
+newrxn.rxnMetaNetXID = regexprep(newrxn.rxnMetaNetXID,'_cv','');
+fclose(fid);
+
+% Change coefficients for reactants:
+for i=1:length(matrix.rxnIDs)
+    if strcmp(matrix.mettype(i),'reactant')
+        matrix.metcoef(i) = matrix.metcoef(i)*-1;
+    end
+end
+
+% Change compartments:
+CONValldata = cat(2,model.compNames,model.comps);
+lbracket    = ' [' ;
+llbracket   = '[';
+rbrackets   = ']';
+space       = ' ';
```

Comment:
this variable is not used and can be removed

Reply:
Addressed.